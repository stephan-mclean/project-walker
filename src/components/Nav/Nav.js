import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styles from "./Nav.module.scss";

const Nav = ({
  onHomeClick,
  onSettingsClick,
  mainBtnIcon,
  onMainBtnClick,
  className
}) => {
  const navClassName = `${
    styles.container
  } secondary-contrast-coloring shadow-fg-tertiary-1-top-25 flex flex--space-evenly flex--center-y ${className ||
    ""}`;

  return (
    <nav className={navClassName}>
      <FontAwesomeIcon icon="home" onClick={onHomeClick} />
      {mainBtnIcon && (
        <FontAwesomeIcon
          icon={mainBtnIcon}
          onClick={onMainBtnClick}
          className="text--xl"
        />
      )}
      <FontAwesomeIcon icon="cog" onClick={onSettingsClick} />
    </nav>
  );
};

Nav.propTypes = {
  onHomeClick: PropTypes.func,
  onSettingsClick: PropTypes.func,
  mainBtnIcon: PropTypes.string,
  onMainBtnClick: PropTypes.func,
  className: PropTypes.string
};

const emptyFn = () => {};
Nav.defaultProps = {
  onHomeClick: emptyFn,
  onSettingsClick: emptyFn,
  onMainBtnClick: emptyFn
};

export default Nav;
