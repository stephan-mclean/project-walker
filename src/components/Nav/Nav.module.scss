.container {
  height: 4rem;
  padding: var(--component-padding);
}
