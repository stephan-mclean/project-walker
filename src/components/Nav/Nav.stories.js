import React, { Fragment } from "react";
import { storiesOf } from "@storybook/react";

import Nav from "./Nav";

storiesOf("Components", module).add("Nav", () => {
  return (
    <Fragment>
      <Nav className="margin-top margin-bottom" />
      <Nav mainBtnIcon="plus-square" />
    </Fragment>
  );
});
