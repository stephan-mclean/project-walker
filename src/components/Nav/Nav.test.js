import React from "react";
import { shallow } from "enzyme";

import Nav from "./Nav";

describe("Testing the Nav component", () => {
  it("Should render the default icons", () => {
    const nav = shallow(<Nav />);

    expect(nav).toMatchSnapshot();
    expect(nav.find('FontAwesomeIcon[icon="home"]').exists()).toBeTruthy();
    expect(nav.find('FontAwesomeIcon[icon="cog"]').exists()).toBeTruthy();
  });

  it("Should respond to the users clicks", () => {
    const onHomeClick = jest.fn();
    const onSettingsClick = jest.fn();

    const nav = shallow(
      <Nav onHomeClick={onHomeClick} onSettingsClick={onSettingsClick} />
    );

    const homeBtn = nav.find('FontAwesomeIcon[icon="home"]');
    homeBtn.simulate("click");
    expect(onHomeClick).toHaveBeenCalled();

    const settingsBtn = nav.find('FontAwesomeIcon[icon="cog"]');
    settingsBtn.simulate("click");
    expect(onSettingsClick).toHaveBeenCalled();
  });

  it("Should render the main button", () => {
    const nav = shallow(<Nav mainBtnIcon="someicon" />);
    expect(nav).toMatchSnapshot();
    expect(nav.find('FontAwesomeIcon[icon="someicon"]').exists()).toBeTruthy();
  });

  it("Should respond to the main button being clicked", () => {
    const onMainBtnClick = jest.fn();
    const nav = shallow(
      <Nav mainBtnIcon="someicon" onMainBtnClick={onMainBtnClick} />
    );
    const mainBtn = nav.find('FontAwesomeIcon[icon="someicon"]');

    mainBtn.simulate("click");
    expect(onMainBtnClick).toHaveBeenCalled();
  });

  it("Should allow passing a custom class", () => {
    const nav = shallow(<Nav className="some-custom-class" />);
    expect(nav).toMatchSnapshot();
    expect(nav.hasClass("some-custom-class")).toBeTruthy();
  });
});
