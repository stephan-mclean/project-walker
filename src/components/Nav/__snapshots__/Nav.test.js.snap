// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Nav component Should allow passing a custom class 1`] = `
<nav
  className="container secondary-contrast-coloring shadow-fg-tertiary-1-top-25 flex flex--space-evenly flex--center-y some-custom-class"
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="home"
    inverse={false}
    listItem={false}
    mask={null}
    onClick={[Function]}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="cog"
    inverse={false}
    listItem={false}
    mask={null}
    onClick={[Function]}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</nav>
`;

exports[`Testing the Nav component Should render the default icons 1`] = `
<nav
  className="container secondary-contrast-coloring shadow-fg-tertiary-1-top-25 flex flex--space-evenly flex--center-y "
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="home"
    inverse={false}
    listItem={false}
    mask={null}
    onClick={[Function]}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="cog"
    inverse={false}
    listItem={false}
    mask={null}
    onClick={[Function]}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</nav>
`;

exports[`Testing the Nav component Should render the main button 1`] = `
<nav
  className="container secondary-contrast-coloring shadow-fg-tertiary-1-top-25 flex flex--space-evenly flex--center-y "
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="home"
    inverse={false}
    listItem={false}
    mask={null}
    onClick={[Function]}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
  <FontAwesomeIcon
    border={false}
    className="text--xl"
    fixedWidth={false}
    flip={null}
    icon="someicon"
    inverse={false}
    listItem={false}
    mask={null}
    onClick={[Function]}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="cog"
    inverse={false}
    listItem={false}
    mask={null}
    onClick={[Function]}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</nav>
`;
